scala/private/phases/phase_compile.bzl

```diff
@@ -308,17 +308,32 @@ def _create_scala_compilation_provider(ctx, ijar, source_jar, deps_providers):
         neverlink = ctx.attr.neverlink,
     )
 
-def _pack_source_jar(ctx, scala_srcs, in_srcjars):
-    output_jar = ctx.outputs.jar
-    source_jar_name = output_jar.basename[:-len(output_jar.extension)] + "-src.jar"
-    output_source_jar = ctx.actions.declare_file(source_jar_name, sibling = output_jar)
-    return java_common.pack_sources(
-        ctx.actions,
-        output_source_jar = output_source_jar,
-        sources = scala_srcs,
-        source_jars = in_srcjars,
-        java_toolchain = find_java_toolchain(ctx, ctx.attr._java_toolchain),
-    )
+def _pack_source_jar(ctx, scala_srcs, input_srcjars):
+    # https://github.com/bazelbuild/bazel/blob/ff6c0333e4f957bb9f7ab5401b01dbf3e9b515b1/src/main/java/com/google/devtools/build/lib/rules/java/JavaInfoBuildHelper.java#L180-L183
+    # java_common.pack_sources checks for no srcs and only a single input jar
+    # if so, it checks that output_source_jar is null
+    # passing that, it will return the input source jar directly
+    # However, pack_sources will FAIL if both output_source_jar and
+    # the deprecated output_jar field are BOTH null
+    # Therefore, we can return the single input jar ourselves
+    if not scala_srcs and len(input_srcjars) == 1:
+        return input_srcjars[0]
+    else:
+        output_jar = ctx.outputs.jar
+        without_ext = output_jar.basename[:-len(output_jar.extension)]
```

Comment:
Not sure if this would make code look better and probably it's a matter of taste but `without_ext` could be calculated as `output_jar.basename[:-len(output_jar.extension) - 1] if output_jar.extension else output_jar.basename`
Please feel free to ignore this comment.

Reply:
Good suggestion! I ended up with (ironically) a longer implementation. Your suggestion made me realize that we shouldn't expect that output_jar *has* an extension. Personally, I think that the ternary `if` is too long for one line, so a mutating `if` is fine.